x-pack/plugins/apm/server/lib/security/get_indices_privileges.test.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Logger } from 'src/core/server';
+import { Setup } from '../helpers/setup_request';
+import { getIndicesPrivileges } from './get_indices_privileges';
+
+describe('getIndicesPrivileges', () => {
+  const indices = {
+    apm_oss: {
+      errorIndices: 'apm-*',
+      metricsIndices: 'apm-*',
+      transactionIndices: 'apm-*',
+      spanIndices: 'apm-*'
+    }
+  };
+  const logger = ({
+    warn: jest.fn()
+  } as unknown) as Logger;
+  it('has privileges when an error code 400 happens while fetching ', async () => {
+    const setup = ({
+      indices,
+      client: {
+        hasPrivileges: () => {
+          const error = {
+            message:
+              'no handler found for uri [/_security/user/_has_privileges]',
+            statusCode: 400
+          };
+          throw error;
+        }
+      }
+    } as unknown) as Setup;
+    const privileges = await getIndicesPrivileges(setup, logger);
+    expect(privileges).toEqual({
+      has_all_requested: true,
+      index: {}
+    });
+  });
+  it('throws any other error other than 400 ', async () => {
+    const setup = ({
+      indices,
+      client: {
+        hasPrivileges: () => {
+          throw new Error('unknow error');
+        }
+      }
+    } as unknown) as Setup;
+    try {
+      await getIndicesPrivileges(setup, logger);
```

Comment:
Having expectations in branches like try/catch is not advised since it can cause the test to be silently skipped (unless you provide `expect.assertions(number)`).

Something like this is less error-prone.
```suggestion
      await expect(getIndicesPrivileges(setup, logger)).rejects.toThrowError('unknow error')
```

https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/no-try-expect.md

Reply:
Oh, I see that `eslint-plugin-jest` is already available in Kibana. We might have to enable it for APM explicitly somehow.

Reply:
Yeah, feel free to add an override to the APM specific rules to try it out there https://github.com/elastic/kibana/blob/master/.eslintrc.js#L518-L520